### ISSUE TITLE
Support for Micrometer's Observation API

### ISSUE THREAD

**Issue by brunobat** [quarkusio/quarkus#29911]
### Description

We have instrumentation comming from many places: OTel Tracing, Micrometer metrics, MP Metrics...
In the future this is likely to grow, meaning we need to create an abstraction that will help to wire many diferent observability related libraries into the main Quarkus libraries.

Dependent on this issue:
https://github.com/quarkusio/quarkus/issues/30712

**Comment by musufyian** [comment]
@brunobat how can I contribute to this initiative ?

**Comment by t1** [comment]
I have the impression that currently everything moves towards OTel... and that's already a pretty generic standard. Do you have more detailed ideas as to how another level of abstraction ontop of this would look like?

**Comment by brunobat** [comment]
Yes @t1. I'm seriously considering this: https://micrometer.io/docs/observation

**Comment by t1** [comment]
Isn't Micrometer Observation on a too high level of abstraction? I.e., it doesn't support the concept of Traces, etc. I'm not sure if that would actually be a step backwards.

**Comment by t1** [comment]
Hmmm... I assume you would use Micrometer Tracing for Tracing, but this is only about Observability. I guess that I got it, now 😁

**Comment by brunobat** [comment]
No, The observation API is totally independent from the actual Micrometer metrics and tracing. That's why it's interesting

**Comment by mederel** [comment]
Hi, I have the impression the OpenTelemetry API (its metrics part if we are talking about that) could be directly the good candidate to fill the position of the abstraction layer. If one wants to continue using micrometer as a "backend" for metrics there is the https://github.com/open-telemetry/opentelemetry-java-contrib/tree/main/micrometer-meter-provider. It would allow to keep both during a transition phase.

**Comment by brunobat** [comment]
Thanks @mederel. The idea here is to instrument libraries in one place and be able to then plug different metrics and tracing implementations... Currently we have to instrument everything with OTel and Micrometer because users can chose to use both

**Comment by kerkhofsd** [comment]
Any update on this topic? 

In the [Quarkus documentation for OpenTelmetry](https://quarkus.io/guides/opentelemetry), the following is mentioned:

> In the future, we plan to bridge current Micrometer metrics to OpenTelemetry and maintain compatibility when possible.

Can this "bridge" be considered part of this scope?

**Comment by brunobat** [comment]
> Can this "bridge" be considered part of this scope?

Yes @kerkhofsd and I plan to work on this until the end of the year.

**Comment by kerkhofsd** [comment]
Great, thanks for the feedback @brunobat !

**Comment by brunobat** [comment]
Feedback is welcome on the existing PR: https://github.com/quarkusio/quarkus/pull/55145
